collectors/proc.plugin/sys_block_zram.c

```diff
@@ -0,0 +1,288 @@
+
+    snprintfz(chart_name, RRD_ID_LENGTH_MAX, "zram_usage.%s", name);
+    d->st_usage = rrdset_create_localhost(
+        "mem"
+        , chart_name
+        , chart_name
+        , name
+        , "mem.zram_usage"
+        , "ZRAM Memory Usage"
+        , "MiB"
+        , PLUGIN_PROC_NAME
+        , PLUGIN_PROC_MODULE_ZRAM_NAME
+        , NETDATA_CHART_PRIO_MEM_ZRAM
+        , update_every
+        , RRDSET_TYPE_AREA);
+    d->rd_compr_data_size = rrddim_add(d->st_usage, "compressed", NULL, 1, 1024 * 1024, RRD_ALGORITHM_ABSOLUTE);
+    d->rd_metadata_size = rrddim_add(d->st_usage, "metadata", NULL, 1, 1024 * 1024, RRD_ALGORITHM_ABSOLUTE);
+
+    snprintfz(chart_name, RRD_ID_LENGTH_MAX, "zram_savings.%s", name);
+    d->st_savings = rrdset_create_localhost(
+        "mem"
+        , chart_name
+        , chart_name
+        , name
+        , "mem.zram_savings"
+        , "ZRAM Memory Savings"
+        , "MiB"
+        , PLUGIN_PROC_NAME
+        , PLUGIN_PROC_MODULE_ZRAM_NAME
+        , NETDATA_CHART_PRIO_MEM_ZRAM_SAVINGS
+        , update_every
+        , RRDSET_TYPE_AREA);
+    d->rd_savings_size = rrddim_add(d->st_savings, "savings", NULL, 1, 1024 * 1024, RRD_ALGORITHM_ABSOLUTE);
+    d->rd_original_size = rrddim_add(d->st_savings, "original", NULL, 1, 1024 * 1024, RRD_ALGORITHM_ABSOLUTE);
+
+    snprintfz(chart_name, RRD_ID_LENGTH_MAX, "zram_ratio.%s", name);
+    d->st_comp_ratio = rrdset_create_localhost(
+        "mem"
+        , chart_name
+        , chart_name
+        , name
+        , "mem.zram_ratio"
+        , "ZRAM Compression Ratio (original to compressed)"
+        , "ratio"
+        , PLUGIN_PROC_NAME
+        , PLUGIN_PROC_MODULE_ZRAM_NAME
+        , NETDATA_CHART_PRIO_MEM_ZRAM_RATIO
+        , update_every
+        , RRDSET_TYPE_LINE);
+    d->rd_comp_ratio = rrddim_add(d->st_comp_ratio, "ratio", NULL, 1, 100, RRD_ALGORITHM_ABSOLUTE);
+
+    snprintfz(chart_name, RRD_ID_LENGTH_MAX, "zram_efficiency.%s", name);
+    d->st_alloc_efficiency = rrdset_create_localhost(
+        "mem"
+        , chart_name
+        , chart_name
+        , name
+        , "mem.zram_efficiency"
+        , "ZRAM Efficiency"
+        , "percentage"
+        , PLUGIN_PROC_NAME
+        , PLUGIN_PROC_MODULE_ZRAM_NAME
+        , NETDATA_CHART_PRIO_MEM_ZRAM_EFFICIENCY
+        , update_every
+        , RRDSET_TYPE_LINE);
+    d->rd_alloc_efficiency = rrddim_add(d->st_alloc_efficiency, "percent", NULL, 1, 10000, RRD_ALGORITHM_ABSOLUTE);
+}
+
+static int init_devices(DICTIONARY *devices, unsigned int zram_id, int update_every) {
+    int count = 0;
+    DIR *dir = opendir("/dev");
+    struct dirent *de;
+    struct stat st;
+    char filename[FILENAME_MAX + 1];
+    procfile *ff = NULL;
+    ZRAM_DEVICE device;
+
+    if (unlikely(!dir))
+        return 0;
+    while ((de = readdir(dir)))
+    {
+        snprintfz(filename, FILENAME_MAX, "/dev/%s", de->d_name);
+        if (unlikely(stat(filename, &st) != 0))
+        {
```

Comment:
Hey, it would be nice to not add another code style

https://docs.netdata.cloud/contributing/#code-style

https://www.kernel.org/doc/html/v4.10/process/coding-style.html#placing-braces-and-spaces